### PR TITLE
Feat: util msg fmt

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
// uniform the err/wrn msg fmt.

### CODE REVIEW

**Comment by Shane-XB-Qian** [comment]
@yegappan ready to merge if ok.

**Comment by andlrc** [comment]
I think this should be merged, as the messages are now easier to digest, as the type is prefixed 👍

File: test/clangd_tests.vim [hunk 1]
```diff
@@ -113,8 +113,8 @@ def g:Test_LspFormat()
 
   # file without an LSP server
   edit a.raku
-  assert_equal(['Error: Language server for "raku" file type is not found'],
```
Comment (yegappan) [inline test/clangd_tests.vim:116]: I would like to merge this PR.  Can you revert the changes to the clangd_tests.vim file that changes assert_equal() to assert_match()?

**Review by yegappan — CHANGES_REQUESTED** [review]

File: test/clangd_tests.vim [hunk 2]
```diff
@@ -109,8 +109,8 @@ def g:Test_LspFormat()
...
-	       execute('LspFormat')->split("\n"))
+  assert_match('Language server for "raku" file type is not found',
```
[the remark this answers is not in the document]
Comment (Shane-XB-Qian) [reply test/clangd_tests.vim:116]: is impossible to be `equal`? it had prefix and type now.

File: test/clangd_tests.vim (hunk 1 above)
Comment (Shane-XB-Qian) [reply test/clangd_tests.vim:116]: is impossible to be `equal`? it had prefix and type now.

File: test/clangd_tests.vim (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shane-XB-Qian) [reply test/clangd_tests.vim:116]: it had uniformed the prefix/type. that's why i made this change. it is impossible to be `equal`.
// wish you're not to against my every PR. -_-#

File: test/clangd_tests.vim (hunk 1 above)
Comment (Shane-XB-Qian) [reply test/clangd_tests.vim:116]: it had uniformed the prefix/type. that's why i made this change. it is impossible to be `equal`.
// wish you're not to against my every PR. -_-#

File: test/clangd_tests.vim (hunk 2 above)
[the remark this answers is not in the document]
Comment (yegappan) [reply test/clangd_tests.vim:116]: What is the new error message with your change?  Why can't assert_equal() be used?  It looks like with your change, the error message text should still remain the same.
